localization/depth_odometry/src/image_features_with_known_correspondences_aligner_depth_odometry.cc

```diff
@@ -125,11 +130,62 @@ ImageFeaturesWithKnownCorrespondencesAlignerDepthOdometry::DepthImageCallback(
     target_landmarks.emplace_back(target_landmark);
   }
 
-  if (target_landmarks.size() < 4) {
-    LogError("DepthImageCallback: Too few points provided, need 4 but given " << target_landmarks.size() << ".");
+  if (target_landmarks.size() < params_.min_num_correspondences) {
+    LogDebug("DepthImageCallback: Too few points provided, need " << params_.min_num_correspondences << " but given "
+                                                                  << target_landmarks.size() << ".");
     return boost::none;
   }
 
+  if (params_.filter_outliers) {
+    std::vector<unsigned char> inliers;
+    std::vector<cv::Point2d> source_points, target_points;
+    // Undistort and store image points as CV points
+    for (int i = 0; i < source_image_points.size(); ++i) {
+      Eigen::Vector2d undistorted_source, undistorted_target;
+      params_.cam_params->Convert<camera::DISTORTED, camera::UNDISTORTED_C>(source_image_points[i],
+                                                                            &undistorted_source);
+      params_.cam_params->Convert<camera::DISTORTED, camera::UNDISTORTED_C>(target_image_points[i],
+                                                                            &undistorted_target);
+      source_points.emplace_back(cv::Point2d(undistorted_source.x(), undistorted_source.y()));
+      target_points.emplace_back(cv::Point2d(undistorted_target.x(), undistorted_target.y()));
+    }
+    cv::Mat intrinsics;
+    cv::eigen2cv(params_.cam_intrinsics, intrinsics);
+    // TODO(rsoussan): Pass iterations limit if OpenCV version upgraded
```

Comment:
did you mean to opencv 4? (we did upgrade)

Reply:
No, I actually meant opencv 4.xx, I'll look it up and revise the comment. It's not high priority because I don't actually use that outlier rejection in practice (it didn't help the results).